### PR TITLE
[RSDK-1008] Support GPIO pins on the NVidia Jetson Orin board

### PR DESCRIPTION
**Pull request by penguinland**
Tried at my desk: with these changes linked into the Viam RDK, I'm able to set the GPIO pins high and low through our app!

Motivation for this is described at https://github.com/periph/host/issues/26: on every other (non-Orin) board, if you enable a GPIO pin, it gets created in `/sys/class/gpio/gpioN` where `N` is the pin's number. but the Orin uses different names, so we need to include those names separately. 

Linking things up: this is the same PR as https://github.com/periph/host/pull/27, but into our own fork instead of the original repo. This is a quick-and-dirty approach to get our Orin prototype up and running. This approach doesn't support muxing the pins to other locations (apologies if that's the wrong phrasing, I only half understand what it means), hence why the other version of this PR might not get accepted into the original repo in the near future. We (or they, if they get there first?) can figure that out later, when our support for the Orin is further along.

#### How the list of names was constructed ####

[Nvidia's documentation](https://developer.nvidia.com/docs/drive/drive-os/archives/6.0.3/linux/sdk/oxy_ex-1/common/topics/sys_components/CalculatingGPIOIndexinLinux32.html) describes what they should be in theory. 

In practice, there are 2 GPIO chips, and you can see their offsets and how many pins they have: 
```
viam@orindevkit2:~$ cd /sys/class/gpio
viam@orindevkit2:/sys/class/gpio$ ls
export  gpiochip316  gpiochip348  unexport
viam@orindevkit2:/sys/class/gpio$ cat gpiochip316/base
316
viam@orindevkit2:/sys/class/gpio$ cat gpiochip316/ngpio 
32
viam@orindevkit2:/sys/class/gpio$ cat gpiochip348/base
348
viam@orindevkit2:/sys/class/gpio$ cat gpiochip348/ngpio
164
```
You can see that they're contiguous with each other: one starts at 316 and has 32 pins, and the other starts at 348 and has 164 pins. You can then enumerate all the pins by running `for i in $(seq 316 511); do echo "$i: $(echo $i > export; ls|grep -v export|grep -v gpiochip3; echo $i > unexport)"; done`. A handful of pins give errors rather than being exported, but all the rest of the names match up with what they're supposed to be in theory.

### CODE REVIEW

File: sysfs/gpio.go [hunk 1]
```diff
@@ -505,6 +509,44 @@ func (d *driverGPIO) parseGPIOChip(path string) error {
...
+	"X.04", "X.05", "X.06", "X.07", "Y.00", "Y.01", "Y.02", "Y.03", "Y.04", "Y.05",
+	"Y.06", "Y.07", "Z.00", "Z.01", "Z.02", "Z.03", "Z.04", "Z.05", "Z.06", "Z.07",
+	"AC.00", "AC.01", "AC.02", "AC.03", "AC.04", "AC.05", "AC.06", "AC.07", "AD.00", "AD.01",
+	"AD.02", "AD.03", "AE.00", "AE.01", "AF.00", "AF.01", "AF.02", "AF.03", "AG.00", "AG.01",
+	"AG.02", "AG.03", "AG.04", "AG.05", "AG.06", "AG.07",
+}
+
+func getSymlinkRoot(boardModel string, pinNumber int) string {
+	if boardModel == "Jetson AGX Orin" {
+		pinName := jetsonOrinAgxPinNames[pinNumber-jetsonOrinAgxOffset]
```
Comment (randhid) [inline sysfs/gpio.go:542]: [minor] we might want to make this a function that takes the map and the board name if we need to add more funky mapping down the line. Non-blocking, and overall lgtm.

**Review by randhid — APPROVED** [review]
lgtm, one suggestion that may make life slightly easier for adding future maps.

**Review by Otterverse — APPROVED** [review]
As discussed, I think this is fine for now. I do think that it would be worth the time to get satisfy upstream devs so they accept this there though. Up to your team to decide if/when to prioritize it, but I'd at least like to see a JIRA ticket for it so it doesn't get forgotten.

File: sysfs/gpio.go (hunk 1 above)
Comment (penguinland) [reply sysfs/gpio.go:542]: Yeah, I think you might be right. but without a second example of a board with nonstandard names, it's unclear to me what the best approach is. I'm inclined to leave as-is and change it when we encounter the next board.

**Pull request merged**